conans/client/conf/detect.py

```diff
@@ -34,6 +34,25 @@ def _gcc_compiler(compiler_exe="gcc"):
             return compiler, installed_version
     except Exception:
         return None
+    
+def _msvc_cl_compiler(compiler_exe="cl"):
+    try:
+        compiler_exe = compiler_exe.strip('"')
+        ret, out = detect_runner(f'"{compiler_exe}" /?')
+        if ret != 0:
+            return None
+        first_line = out.splitlines()[0]
+        if not "Microsoft" in first_line:
+            return None
+        compiler = "msvc"
+        version_regex = re.search(r"(?P<major>[0-9]+)\.(?P<minor>[0-9]+)\.([0-9]+)\.?([0-9]+)?", first_line)
```

Comment:
Maybe the last optional fourth group can just be dropped, no need to even match it?

Reply:
I considered that - I added the last option groups since so far the version seems to always have 3 or 4 components - figured it's safer to get a match of the longest string that looks like a version? In case something ever changes in that output :/ but happy to drop this to two

Reply:
Not necessary, it is fine, I guess it shouldn't be a problem